daylay92/MarkdownToMarkUpConverter
# Implement markdown conversion

#### What does this PR do?
Implement the algorithm for converting markdown to markup
#### Description of Task to be completed?
- Implement a regex class library with each property, a representation of a regex syntax that matches certain markdown syntax.
- Add methods to the regex class for manipulating and converting markdown into markup by utilizing the pre-defined properties.
- Add function to the entry point `app.js` that serves as an API that reads a markdown file from the `./markdown` directory, delegates its conversion to the regex class library methods, and writes a corresponding equivalent in markup format into the `./markup` directory.

#### How should this be manually tested?
- To run this project, install it locally using npm:
```
$ cd MarkdownToMarkUpConverter
$ mkdir markdown
$ npm install
$ npm run build
```
* After setup is complete, Open the project folder and navigate to the *markdown* sub-folder.
* Move the Markdown file you wish to convert into the *markdown* sub-folder.
* Return to your command line palette and be sure you are in the root of the project.
* From the root of the project run the command `npm run convert [filename.extension]` Example : `npm run convert markdown.md`
* Check the command line palette for the output. Also navigate to the *markup* sub-folder to get the markup file created.
